Extensions/Optimizely/Src/Tasks/OptimizelyX/operations/TaskOperation.ts

```diff
@@ -0,0 +1,196 @@
+          await this.sleep(10000);
+          counter--;
+      }
+
+      console.log(tl.loc("ABTestStartedSuccessfully", experimentName));
+      console.log(tl.loc("VisitOptimizelyPage") + ` https://app.optimizely.com/v2/projects/${projectId}/experiments`);
+  }
+
+  public async stopABTest(experiment: IOptimizelyAlphaBetaTest): Promise<void> {
+      let environmentName: string = this.taskInputs.getEnvironmentName();
+      let projectId: string = this.taskInputs.getProjectId();
+
+      if(this.taskInputs.isWebProject()) {
+          experiment.status = OptimizelyExperimentStatus.Paused;
+      } else {
+          let environment: IOptimizelyEnvironment = await this.optimizelyClient.getEnvironment(projectId, environmentName);
+          if (environment == null) {
+              throw tl.loc("EnvironmentNotFound", environmentName);
+          }
+
+          Object.keys(experiment.environments).forEach((value) => {
+              if (value.toLowerCase() == environmentName.toLowerCase()) {
+                  experiment.environments[value].status = OptimizelyExperimentStatus.Paused;
+              }
+          });
+      }
+
+      let experimentName = experiment.name;
+
+      await this.optimizelyClient.updateExperiment(experiment.id, experiment);
+      console.log(tl.loc("ExperimentSuccessfullyPaused", experimentName));
+  }
+
+  public updateTrafficVariation(experiment: IOptimizelyAlphaBetaTest): IOptimizelyAlphaBetaTest {
+      let totalTrafficParameter : string = this.taskInputs.getTotalTraffic();
+      let totalTraffic: number = Number.parseInt(totalTrafficParameter);
+
+      if (Number.isNaN(totalTraffic)) {
+          throw tl.loc("TrafficValueNotValid", totalTrafficParameter);
+      }
+
+      if (totalTraffic > 100) {
+          throw tl.loc("TrafficValueCantExceed", totalTraffic);
+      }
+
+      let holdBack: number = 10000 - (totalTraffic * 100);
+      experiment.holdback = holdBack;
+      let trafficByVariationKeyInput = this.taskInputs.getTrafficByVariation();
+      let trafficVariations = this.parseTrafficByVariation(trafficByVariationKeyInput);
+
+      for (let i: number = 0; i < experiment.variations.length; i++) {
+          let variationKey: string = (this.taskInputs.isWebProject()) ? experiment.variations[i].name : experiment.variations[i].key;
+          variationKey = variationKey.toLowerCase();
+          if (!trafficVariations.has(variationKey)) {
+              throw tl.loc("TrafficVariationKeyNotDefined", variationKey);
+          }
+          experiment.variations[i].weight = trafficVariations.get(variationKey);
+      }
+
+      return experiment;
+  }
+
+  public async getAudienceCondition(projectId: string): Promise<string> {
+      let audienceInput: string = this.taskInputs.getAudience();
+      let audience: string = null;
+
+      if (audienceInput == "" || audienceInput == null) {
+          audience = "everyone";
+      } else {
+          let result: Array<any> = new Array<any>();
```

Comment:
right now we use an "or" over the audience list eg. set the audience as "North America or Europe", but it could just as easily be an "and" as well  eg. "north america AND age is 18". I think PM's should take a call how audience should be handled.